library/common.h

```diff
@@ -68,6 +74,19 @@ extern void (*mbedtls_test_hook_test_fail)( const char * test, int line, const c
  */
 #define MBEDTLS_ALLOW_PRIVATE_ACCESS
 
+/** Detect architectures where unaligned memory accesses are safe and performant.
+ *
+ * This list is incomplete.
+ */
+#if (defined(__i386__) || defined(__amd64__) || defined( __x86_64__) \
+    || defined(__ARM_FEATURE_UNALIGNED) \
+    || defined(__aarch64__) \
+    || defined(__ARM_ARCH_8__) || defined(__ARM_ARCH_8A__) || defined(__ARM_ARCH_8M__) \
+    || defined(__ARM_ARCH_7A__)) \
+    && (!(defined(__has_feature) && __has_feature(undefined_behavior_sanitizer)))
```

Comment:
“When checking for undefined behavior, disable the undefined behavior.”

This doesn't seem right to me. Are you sure that the complaints from UBSan aren't warranted? Just because the CPU supports misaligned accesses doesn't mean the compiler translates them as expected.

Reply:
That is a good point, I had assumed that UBSan was checking beyond the standard, but apparently in C99 it's UB to convert a pointer if it results in misalignment. I think `__attribute__ ((packed))` may be the better solution.